lib/xenops_server.ml

```diff
@@ -2717,12 +2717,13 @@ module VBD = struct
   let string_of_id (a, b) = a ^ "." ^ b
 
   let add' x =
-    debug "VBD.add %s %s" (string_of_id x.id) (Jsonrpc.to_string (rpc_of t x)) ;
-    (* Only if the corresponding VM actually exists *)
+    let dom0 = Inventory.lookup Inventory._control_domain_uuid in
```

Comment:
Let's not do this lookup for every `add` call, because dom0 won't change. Have a function that caches it in a ref.

Reply:
Yes, see https://github.com/xapi-project/xenopsd/pull/755#discussion_r628331582

Reply:
I've implemented the optimization